netket/operator/_hamiltonian.py

```diff
@@ -91,17 +91,18 @@ def __rmul__(self, other):
         return self.to_local_operator().__rmul__(other)
 
     def _op__matmul__(self, other):
+        if hasattr(other, "to_local_operator"):
+            other = other.to_local_operator()
         return self.to_local_operator().__matmul__(other)
 
     def _op__rmatmul__(self, other):
         if self == other and self.is_hermitian:
             return Squared(self)
+        if hasattr(other, "to_local_operator"):
```

Comment:
why you don't call squared here on the local_operator ?

Reply:
I don't understand your comment?
anyhow the check returning squared here could be removed because it was redundant